source/adapters/cuda/enqueue.cpp

```diff
@@ -530,22 +530,41 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueKernelLaunchCustomExp(
   }
 
   std::vector<CUlaunchAttribute> launch_attribute(numPropsInLaunchPropList);
+  bool has_property_cluster_launch = false;
+
   for (uint32_t i = 0; i < numPropsInLaunchPropList; i++) {
```

Comment:
You may as well move this whole for loop down past line 594 where cuFunc is defined, then you can move
```
UR_CHECK_ERROR(cuFuncSetAttribute(
          CuFunc, CU_FUNC_ATTRIBUTE_NON_PORTABLE_CLUSTER_SIZE_ALLOWED, 1));
```
inside `UR_EXP_LAUNCH_PROPERTY_ID_CLUSTER_DIMENSION` case, then you don't need `has_property_cluster_launch` at all.

Reply:
Done